### PR TITLE
Refactor office 365 integration tests

### PR DESCRIPTION
**Pull request by mjcr99**
|Related issue|
|---|
|https://github.com/wazuh/wazuh/issues/17612|

## Description

As commented in issue https://github.com/wazuh/wazuh/issues/17427, we are migrating the ITs from QA repository to the core repository.

By this module, we add the Office365 integration tests.

## Execution logs
```
root@agent3-ubu22:/vagrant/Issue17172/wazuh/tests/integration/test_office365/test_configuration# python3 -m pytest -vv  test_invalid.py 
==================================================================== test session starts ====================================================================
platform linux -- Python 3.10.6, pytest-7.1.2, pluggy-1.0.0 -- /usr/bin/python3
cachedir: .pytest_cache
metadata: {'Python': '3.10.6', 'Platform': 'Linux-5.15.0-25-generic-x86_64-with-glibc2.35', 'Packages': {'pytest': '7.1.2', 'pluggy': '1.0.0'}, 'Plugins': {'html': '3.1.1', 'metadata': '3.0.0'}}
rootdir: /vagrant/Issue17172/wazuh/tests/integration, configfile: pytest.ini
plugins: html-3.1.1, metadata-3.0.0
collected 8 items                                                                                                                                           

test_invalid.py::test_invalid[Invalid_Enabled] PASSED                                                                                                 [ 12%]
test_invalid.py::test_invalid[Invalid_Only_Future_Events] PASSED                                                                                      [ 25%]
test_invalid.py::test_invalid[Invalid_Interval] PASSED                                                                                                [ 37%]
test_invalid.py::test_invalid[Invalid_Curl_Max_Size] PASSED                                                                                           [ 50%]
test_invalid.py::test_invalid[Empty_Tenant_Id] PASSED                                                                                                 [ 62%]
test_invalid.py::test_invalid[Empty_Client_Id] PASSED                                                                                                 [ 75%]
test_invalid.py::test_invalid[Empty_Client_Secret] PASSED                                                                                             [ 87%]
test_invalid.py::test_invalid[Empty_Suscription] PASSED                                                                                               [100%]

=============================================================== 8 passed in 100.35s (0:01:40) =============================================================== 
```

## Testing performed

| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @mjcr99  (Developer)  | test_integration    | | [:green_circle::green_circle::green_circle:](https://github.com/wazuh/wazuh/files/11882532/report.zip) |  Ubuntu Manager       | https://github.com/wazuh/wazuh/pull/17666/commits/23f23296ebd27e1cb8e4d554e86365660666174f | Nothing to highlight |
| @Deblintrake09   (Reviewer)   | test_integration          | |  []()  |Ubuntu Manager        | | |

### CODE REVIEW

**Review by Deblintrake09 — DISMISSED** [review]
LGTM!

File: .github/workflows/integration-tests-office365-tier-0-1-lin.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+          BRANCH_NAME=main
+          if [ "X`git ls-remote --heads https://github.com/wazuh/qa-integration-framework.git ${GITHUB_HEAD_REF}`" != "X" ]; then BRANCH_NAME=${GITHUB_HEAD_REF}; fi
+          git clone -b $BRANCH_NAME --single-branch https://github.com/wazuh/qa-integration-framework.git
+          sudo pip install qa-integration-framework/
+          sudo rm -rf qa-integration-framework/
+      # Run office365 integration tests.
+      - name: Run office365 integration tests
+        run: |
+          cd tests/integration
+          sudo python -m pytest --tier 0 --tier 1 test_office365/
```
Comment (TomasTurina) [inline .github/workflows/integration-tests-office365-tier-0-1-lin.yml:73]: Add new line at the end.

File: .github/workflows/integration-tests-office365-tier-0-1-win.yml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+              $BRANCH_NAME = $env:GITHUB_HEAD_REF
+          }
+          git clone -b $BRANCH_NAME --single-branch https://github.com/wazuh/qa-integration-framework.git
+          pip install qa-integration-framework/
+          rm qa-integration-framework/ -r -force
+      # Run office365 integration tests.
+      - name: Run office365 integration tests
+        run: |
+          cd C:\wazuh\tests\integration
+          python -m pytest --tier 0 --tier 1 test_office365\
```
Comment (TomasTurina) [inline .github/workflows/integration-tests-office365-tier-0-1-win.yml:88]: Add new line at the end.

File: tests/integration/test_office365/conftest.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import pytest
+
+from wazuh_testing.constants.paths.logs import WAZUH_LOG_PATH
+from wazuh_testing.modules.modulesd import patterns
+from wazuh_testing.tools.monitors.file_monitor import FileMonitor
+from wazuh_testing.utils import callbacks
+
+@pytest.fixture()
+def wait_for_office365_start():
+    # Wait for module github starts
```
Comment (TomasTurina) [inline tests/integration/test_office365/conftest.py:14]:
```suggestion
    # Wait for module office365 starts
```

File: tests/integration/test_office365/test_configuration/test_invalid.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+from wazuh_testing.utils.configuration import get_test_cases_data
+from wazuh_testing.utils.configuration import load_configuration_template
+from wazuh_testing.utils import callbacks
+from . import CONFIGS_PATH, TEST_CASES_PATH
+
+# Marks
+pytestmark = pytest.mark.tier(level=0)
+
+# variables
+force_restart_after_restoring = True
```
Comment (TomasTurina) [inline tests/integration/test_office365/test_configuration/test_invalid.py:62]: Is this necessary?

Comment (mjcr99) [reply tests/integration/test_office365/test_configuration/test_invalid.py]: Fixed with https://github.com/wazuh/wazuh/pull/17666/commits/af08de8e3f7e724dfd741d9ca760e8cdc11aeb42

File: tests/integration/test_office365/conftest.py (hunk 3 above)
Comment (mjcr99) [reply tests/integration/test_office365/conftest.py]: Fixed with https://github.com/wazuh/wazuh/pull/17666/commits/af08de8e3f7e724dfd741d9ca760e8cdc11aeb42

File: .github/workflows/integration-tests-office365-tier-0-1-lin.yml (hunk 1 above)
Comment (mjcr99) [reply .github/workflows/integration-tests-office365-tier-0-1-lin.yml]: Fixed with https://github.com/wazuh/wazuh/pull/17666/commits/af08de8e3f7e724dfd741d9ca760e8cdc11aeb42

**Review by TomasTurina — APPROVED** [review]

**Pull request merged**